### PR TITLE
TMA-832: support for pluggable component process type

### PR DESCRIPTION
**Pull request by kubamahnert**

### CODE REVIEW

File: lib/gooddata/models/process.rb [hunk 1]
```diff
@@ -209,6 +209,48 @@ def deploy_from_appstore(path, options = { :client => GoodData.client, :project
...
+              name: component_name,
+              version: component_version
+            }
+          }
+        }
+
+        data[:process][:component][:earlyAccessVersion] = options[:early_access_version] if options[:early_access_version]
+        data[:process][:component][:configLocation] = options[:config_location] if options[:config_location]
+
+        res =
```
Comment (yenkins) [inline lib/gooddata/models/process.rb]: Identical code found in process.rb:120, process.rb:194, process.rb:240 (mass = 198)

File: lib/gooddata/models/process.rb [hunk 2]
```diff
@@ -209,6 +209,48 @@ def deploy_from_appstore(path, options = { :client => GoodData.client, :project
...
+          if process_id.nil?
+            client.post("/gdc/projects/#{project.pid}/dataload/processes", data)
+          else
+            client.put("/gdc/projects/#{project.pid}/dataload/processes/#{process_id}", data)
+          end
+
+        if res.keys.first == 'asyncTask'
```
Comment (yenkins) [inline lib/gooddata/models/process.rb]: Identical code found in process.rb:203, process.rb:247 (mass = 92)

File: lib/gooddata/models/process.rb [hunk 3]
```diff
@@ -209,6 +209,48 @@ def deploy_from_appstore(path, options = { :client => GoodData.client, :project
...
+            name: deploy_name,
+            type: 'COMPONENT',
+            component: {
+              name: component_name,
+              version: component_version
+            }
+          }
+        }
+
+        data[:process][:component][:earlyAccessVersion] = options[:early_access_version] if options[:early_access_version]
```
Comment (yenkins) [inline lib/gooddata/models/process.rb]: Calls 'data[:process]' 2 times (DuplicateMethodCall)

Comment (yenkins) [inline lib/gooddata/models/process.rb]: Calls 'data[:process][:component]' 2 times (DuplicateMethodCall)

Comment (yenkins) [inline lib/gooddata/models/process.rb]: Calls 'options[:early_access_version]' 2 times (DuplicateMethodCall)

File: lib/gooddata/models/process.rb [hunk 4]
```diff
@@ -209,6 +209,48 @@ def deploy_from_appstore(path, options = { :client => GoodData.client, :project
...
+        data[:process][:component][:configLocation] = options[:config_location] if options[:config_location]
```
Comment (yenkins) [inline lib/gooddata/models/process.rb]: Calls 'options[:config_location]' 2 times (DuplicateMethodCall)

File: lib/gooddata/models/process.rb [hunk 5]
```diff
@@ -209,6 +209,48 @@ def deploy_from_appstore(path, options = { :client => GoodData.client, :project
...
+
+        res =
+          if process_id.nil?
+            client.post("/gdc/projects/#{project.pid}/dataload/processes", data)
```
Comment (yenkins) [inline lib/gooddata/models/process.rb]: Calls 'project.pid' 2 times (DuplicateMethodCall)

File: lib/gooddata/models/process.rb [hunk 6]
```diff
@@ -209,6 +209,48 @@ def deploy_from_appstore(path, options = { :client => GoodData.client, :project
         process
       end
 
+      def deploy_component(options = { :client => GoodData.client, :project => GoodData.project })
+        client, project = GoodData.get_client_and_project(options)
+
+        deploy_name = options[:name]
```
Comment (yenkins) [inline lib/gooddata/models/process.rb]: Refers to 'options' more than self (maybe move it to another class?) (FeatureEnvy)

File: lib/gooddata/models/process.rb [hunk 7]
```diff
@@ -209,6 +209,48 @@ def deploy_from_appstore(path, options = { :client => GoodData.client, :project
...
+        fail ArgumentError, 'options[:name] can not be nil or empty!' if deploy_name.nil? || deploy_name.empty?
```
Comment (yenkins) [inline lib/gooddata/models/process.rb]: Performs a nil-check (NilCheck)

Comment (yenkins) [inline lib/gooddata/models/process.rb]: Tests 'deploy_name.nil? || deploy_name.empty?' at least 3 times (RepeatedConditional)

File: lib/gooddata/models/process.rb [hunk 8]
```diff
@@ -209,6 +209,48 @@ def deploy_from_appstore(path, options = { :client => GoodData.client, :project
...
+              version: component_version
+            }
+          }
+        }
+
+        data[:process][:component][:earlyAccessVersion] = options[:early_access_version] if options[:early_access_version]
+        data[:process][:component][:configLocation] = options[:config_location] if options[:config_location]
+
+        res =
+          if process_id.nil?
```
Comment (yenkins) [inline lib/gooddata/models/process.rb]: Tests 'process_id.nil?' at least 3 times (RepeatedConditional)

File: lib/gooddata/models/process.rb [hunk 9]
```diff
@@ -209,6 +209,48 @@ def deploy_from_appstore(path, options = { :client => GoodData.client, :project
         process
       end
 
+      def deploy_component(options = { :client => GoodData.client, :project => GoodData.project })
```
Comment (yenkins) [inline lib/gooddata/models/process.rb]: Has approx 15 statements (TooManyStatements)

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/55/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/868/ : FAILURE in 44s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1136/ : SUCCESS in 2m 10s

File: lib/gooddata/models/process.rb [hunk 10]
```diff
@@ -209,6 +211,51 @@ def deploy_from_appstore(path, options = { :client => GoodData.client, :project
...
+              name: component_name,
+              version: component_version
+            }
+          }
+        }
+
+        data[:process][:component][:earlyAccessVersion] = options[:early_access_version] if options[:early_access_version]
+        data[:process][:component][:configLocation] = options[:config_location] if options[:config_location]
+
+        res =
```
Comment (yenkins) [inline lib/gooddata/models/process.rb]: Identical code found in process.rb:122, process.rb:196, process.rb:245 (mass = 198)

File: lib/gooddata/models/process.rb [hunk 11]
```diff
@@ -209,6 +211,51 @@ def deploy_from_appstore(path, options = { :client => GoodData.client, :project
...
+          if process_id.nil?
+            client.post("/gdc/projects/#{project.pid}/dataload/processes", data)
+          else
+            client.put("/gdc/projects/#{project.pid}/dataload/processes/#{process_id}", data)
+          end
+
+        if res.keys.first == 'asyncTask'
```
Comment (yenkins) [inline lib/gooddata/models/process.rb]: Identical code found in process.rb:205, process.rb:252 (mass = 92)

File: lib/gooddata/models/process.rb [hunk 12]
```diff
@@ -209,6 +211,51 @@ def deploy_from_appstore(path, options = { :client => GoodData.client, :project
         process
       end
 
+      def deploy_component(options = { :client => GoodData.client, :project => GoodData.project })
```
Comment (yenkins) [inline lib/gooddata/models/process.rb]: Has approx 17 statements (TooManyStatements)

File: lib/gooddata/models/project.rb [hunk 13]
```diff
@@ -264,6 +264,9 @@ def transfer_processes(from_project, to_project, options = {})
           to_process = if process.path
                          to_process.delete if to_process
                          GoodData::Process.deploy_from_appstore(process.path, name: process.name, client: to_project.client, project: to_project)
+                       elsif process.component
```
Comment (yenkins) [inline lib/gooddata/models/project.rb]: Calls 'process.component' 3 times (DuplicateMethodCall)

Comment (yenkins) [inline lib/gooddata/models/project.rb]: Refers to 'process' more than self (maybe move it to another class?) (FeatureEnvy)

File: lib/gooddata/models/project.rb [hunk 14]
```diff
@@ -264,6 +264,9 @@ def transfer_processes(from_project, to_project, options = {})
...
+                         to_process.delete if to_process
+                         GoodData::Process.deploy_component(name: process.name, component_name: process.component['name'], component_version: process.component['version'], type: process.type, project: to_project, client: to_project.client)
```
Comment (yenkins) [inline lib/gooddata/models/project.rb]: Calls 'process.name' 8 times (DuplicateMethodCall)

Comment (yenkins) [inline lib/gooddata/models/project.rb]: Calls 'process.type' 4 times (DuplicateMethodCall)

Comment (yenkins) [inline lib/gooddata/models/project.rb]: Calls 'to_project.client' 2 times (DuplicateMethodCall)

Comment (yenkins) [inline lib/gooddata/models/project.rb]: Metrics/LineLength: Line is too long. [239/180]

File: lib/gooddata/models/project.rb [hunk 15]
```diff
@@ -264,6 +264,9 @@ def transfer_processes(from_project, to_project, options = {})
           to_process = if process.path
                          to_process.delete if to_process
                          GoodData::Process.deploy_from_appstore(process.path, name: process.name, client: to_project.client, project: to_project)
+                       elsif process.component
+                         to_process.delete if to_process
```
Comment (yenkins) [inline lib/gooddata/models/project.rb]: Calls 'to_process.delete' 2 times (DuplicateMethodCall)

Comment (yenkins) [inline lib/gooddata/models/project.rb]: Tests 'to_process' at least 4 times (RepeatedConditional)

File: lib/gooddata/models/process.rb [hunk 16]
```diff
@@ -91,6 +91,8 @@ def upload_package(path, files_to_exclude, opts = { :client => GoodData.connecti
       # @option options [String] :process_id ID of a process to be redeployed (do not set if you want to create a new process)
       # @option options [Boolean] :verbose (false) Switch on verbose mode for detailed logging
       def deploy(path, options = { :client => GoodData.client, :project => GoodData.project })
+        return deploy_component(path.merge options) if path.kind_of? Hash and path[:component_name]
```
Comment (yenkins) [inline lib/gooddata/models/process.rb]: Style/NestedParenthesizedCalls: Add parentheses to nested method call `path.merge options`.

Comment (yenkins) [inline lib/gooddata/models/process.rb]: Style/ClassCheck: Prefer `Object#is_a?` over `Object#kind_of?`.

Comment (yenkins) [inline lib/gooddata/models/process.rb]: Style/AndOr: Use `&&` instead of `and`.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/58/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/870/ : FAILURE in 54s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1138/ : SUCCESS in 44s

File: lib/gooddata/models/process.rb [hunk 17]
```diff
@@ -209,6 +211,33 @@ def deploy_from_appstore(path, options = { :client => GoodData.client, :project
...
+        data[:process][:component] = GoodData::Helpers.symbolize_keys(data[:process][:component]).select { |k| [:name, :version].include? k }
+
+        # TODO: we should not have to do this
+        client.post "/gdc/projects/#{project.pid}/projectFeatureFlags",
+                    featureFlag: {
+                      key: 'enableEtlComponent',
+                      value: true
+                    }
+
+        res =
```
Comment (yenkins) [inline lib/gooddata/models/process.rb]: Identical code found in process.rb:122, process.rb:196, process.rb:228 (mass = 198)

File: lib/gooddata/models/process.rb [hunk 18]
```diff
@@ -209,6 +211,33 @@ def deploy_from_appstore(path, options = { :client => GoodData.client, :project
...
+          if process_id.nil?
+            client.post("/gdc/projects/#{project.pid}/dataload/processes", data)
+          else
+            client.put("/gdc/projects/#{project.pid}/dataload/processes/#{process_id}", data)
+          end
+        if res.keys.first == 'asyncTask'
```
Comment (yenkins) [inline lib/gooddata/models/process.rb]: Identical code found in process.rb:205, process.rb:234 (mass = 92)

File: lib/gooddata/models/process.rb [hunk 19]
```diff
@@ -209,6 +211,33 @@ def deploy_from_appstore(path, options = { :client => GoodData.client, :project
         process
       end
 
+      def deploy_component(data, options = { :client => GoodData.client, :project => GoodData.project })
+        client, project = GoodData.get_client_and_project(options)
+        process_id = data[:process_id]
+        data = { process: data } unless data[:process]
```
Comment (yenkins) [inline lib/gooddata/models/process.rb]: Calls 'data[:process]' 4 times (DuplicateMethodCall)

File: lib/gooddata/models/process.rb [hunk 20]
```diff
@@ -209,6 +211,33 @@ def deploy_from_appstore(path, options = { :client => GoodData.client, :project
...
+        data[:process ] = GoodData::Helpers.symbolize_keys(data[:process]).select { |k| [:type, :name, :component].include? k }
+        data[:process][:component] = GoodData::Helpers.symbolize_keys(data[:process][:component]).select { |k| [:name, :version].include? k }
+
+        # TODO: we should not have to do this
+        client.post "/gdc/projects/#{project.pid}/projectFeatureFlags",
```
Comment (yenkins) [inline lib/gooddata/models/process.rb]: Calls 'project.pid' 3 times (DuplicateMethodCall)

File: lib/gooddata/models/process.rb [hunk 21]
```diff
@@ -209,6 +211,33 @@ def deploy_from_appstore(path, options = { :client => GoodData.client, :project
...
+                    featureFlag: {
+                      key: 'enableEtlComponent',
+                      value: true
+                    }
+
+        res =
+          if process_id.nil?
```
Comment (yenkins) [inline lib/gooddata/models/process.rb]: Performs a nil-check (NilCheck)

Comment (yenkins) [inline lib/gooddata/models/process.rb]: Tests 'process_id.nil?' at least 3 times (RepeatedConditional)

File: lib/gooddata/models/process.rb [hunk 22]
```diff
@@ -209,6 +211,33 @@ def deploy_from_appstore(path, options = { :client => GoodData.client, :project
         process
       end
 
+      def deploy_component(data, options = { :client => GoodData.client, :project => GoodData.project })
```
Comment (yenkins) [inline lib/gooddata/models/process.rb]: Has approx 12 statements (TooManyStatements)

File: lib/gooddata/models/process.rb [hunk 23]
```diff
@@ -209,6 +211,33 @@ def deploy_from_appstore(path, options = { :client => GoodData.client, :project
...
+        client, project = GoodData.get_client_and_project(options)
+        process_id = data[:process_id]
+        data = { process: data } unless data[:process]
+        data[:process ] = GoodData::Helpers.symbolize_keys(data[:process]).select { |k| [:type, :name, :component].include? k }
```
Comment (yenkins) [inline lib/gooddata/models/process.rb]: Has the variable name 'k' (UncommunicativeVariableName)

Comment (yenkins) [inline lib/gooddata/models/process.rb]: Style/SpaceInsideBrackets: Space inside square brackets detected.

File: spec/integration/models/process_spec.rb [hunk 24]
```diff
@@ -0,0 +1,35 @@
...
+  describe '.deploy_component' do
+    let(:name) { 'test component' }
+
+    it 'deploys etl pluggable component' do
+      component_data = {
+        name: name,
+        type: :etl,
+        component: {
+          name: 'gdc-etl-sql-executor',
+          version: '1',
```
Comment (yenkins) [inline spec/integration/models/process_spec.rb]: Style/TrailingCommaInLiteral: Avoid comma after the last item of a hash.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/60/ : SUCCESS in 10s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/872/ : FAILURE in 2m 14s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1140/ : SUCCESS in 1m 54s

File: lib/gooddata/models/process.rb [hunk 25]
```diff
@@ -191,26 +176,43 @@ def deploy_from_appstore(path, options = { :client => GoodData.client, :project
           }
         }
 
+        save(data, options)
+      end
+
+      def deploy_component(data, options = { :client => GoodData.client, :project => GoodData.project })
+        client, project = GoodData.get_client_and_project(options)
+        data = { process: data } unless data[:process]
```
Comment (yenkins) [inline lib/gooddata/models/process.rb]: Calls 'data[:process]' 4 times (DuplicateMethodCall)

Comment (yenkins) [inline lib/gooddata/models/process.rb]: Refers to 'data' more than self (maybe move it to another class?) (FeatureEnvy)

File: lib/gooddata/models/process.rb [hunk 26]
```diff
@@ -191,26 +176,43 @@ def deploy_from_appstore(path, options = { :client => GoodData.client, :project
           }
         }
 
+        save(data, options)
+      end
+
+      def deploy_component(data, options = { :client => GoodData.client, :project => GoodData.project })
```
Comment (yenkins) [inline lib/gooddata/models/process.rb]: Has approx 8 statements (TooManyStatements)

File: lib/gooddata/models/process.rb [hunk 27]
```diff
@@ -191,26 +176,43 @@ def deploy_from_appstore(path, options = { :client => GoodData.client, :project
...
+                      key: 'enableEtlComponent',
+                      value: true
+                    }
+
+       save(data, options)
+      end
+
+      private
+
+      def save(data, options = { client: GoodData.client, project: GoodData.project })
```
Comment (yenkins) [inline lib/gooddata/models/process.rb]: Has approx 6 statements (TooManyStatements)

File: lib/gooddata/models/process.rb [hunk 28]
```diff
@@ -191,26 +176,43 @@ def deploy_from_appstore(path, options = { :client => GoodData.client, :project
           }
         }
 
+        save(data, options)
+      end
+
+      def deploy_component(data, options = { :client => GoodData.client, :project => GoodData.project })
+        client, project = GoodData.get_client_and_project(options)
+        data = { process: data } unless data[:process]
+        data[:process ] = GoodData::Helpers.symbolize_keys(data[:process]).select { |k| [:type, :name, :component].include? k }
```
Comment (yenkins) [inline lib/gooddata/models/process.rb]: Has the variable name 'k' (UncommunicativeVariableName)

Comment (yenkins) [inline lib/gooddata/models/process.rb]: Style/SpaceInsideBrackets: Space inside square brackets detected.

File: lib/gooddata/models/process.rb [hunk 29]
```diff
@@ -191,26 +176,43 @@ def deploy_from_appstore(path, options = { :client => GoodData.client, :project
...
+        data[:process][:component] = GoodData::Helpers.symbolize_keys(data[:process][:component]).select { |k| [:name, :version].include? k }
+
+        # TODO: we should not have to do this
+        client.post "/gdc/projects/#{project.pid}/projectFeatureFlags",
+                    featureFlag: {
+                      key: 'enableEtlComponent',
+                      value: true
+                    }
+
+       save(data, options)
```
Comment (yenkins) [inline lib/gooddata/models/process.rb]: Style/IndentationConsistency: Inconsistent indentation detected.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/69/ : SUCCESS in 9s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/881/ : FAILURE in 41s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1149/ : SUCCESS in 34s

File: lib/gooddata/models/process.rb [hunk 30]
```diff
@@ -91,6 +91,8 @@ def upload_package(path, files_to_exclude, opts = { :client => GoodData.connecti
       # @option options [String] :process_id ID of a process to be redeployed (do not set if you want to create a new process)
       # @option options [Boolean] :verbose (false) Switch on verbose mode for detailed logging
       def deploy(path, options = { :client => GoodData.client, :project => GoodData.project })
+        return deploy_component(path, options) if path.is_a? Hash && path[:component]
```
Comment (yenkins) [inline lib/gooddata/models/process.rb]: Lint/RequireParentheses: Use parentheses in the method call to avoid confusion about precedence.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/71/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/883/ : FAILURE in 34s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1151/ : SUCCESS in 37s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/72/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/884/ : SUCCESS in 38s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1152/ : SUCCESS in 32s

File: lib/gooddata/models/process.rb [hunk 31]
```diff
@@ -191,26 +176,43 @@ def deploy_from_appstore(path, options = { :client => GoodData.client, :project
...
+        save(data, options)
+      end
+
+      def deploy_component(data, options = { :client => GoodData.client, :project => GoodData.project })
+        client, project = GoodData.get_client_and_project(options)
+        data = { process: data } unless data[:process]
+        data[:process] = GoodData::Helpers.symbolize_keys(data[:process]).select { |k| [:type, :name, :component].include? k }
+        data[:process][:component] = GoodData::Helpers.symbolize_keys(data[:process][:component]).select { |k| [:name, :version].include? k }
+
+        # TODO: we should not have to do this
```
Comment (panjan) [inline lib/gooddata/models/process.rb]: What exactly? And why do we have to do it?

File: lib/gooddata/models/project.rb [hunk 32]
```diff
@@ -404,6 +407,9 @@ def transfer_schedules(from_project, to_project)
               (schedule_spec[:process_id] == local.process_id) && (schedule.name == schedule_spec[:name])
             end
 
+            # processes without schedules
+            # next unless remote_process && process_spec
```
Comment (panjan) [inline lib/gooddata/models/project.rb]: Some code commented out. ⬆️

File: lib/gooddata/models/project.rb (hunk 13 above)
Comment (panjan) [inline lib/gooddata/models/project.rb]: Should we cover this condition branch with a unit test?

File: lib/gooddata/models/process.rb (hunk 26 above)
Comment (panjan) [inline lib/gooddata/models/process.rb]: Should we cover this method with unit tests?

**Comment by Hahihula** [comment]
ok to merge

**Comment by gdgate** [comment]
Build in pipeline check aborted.

File: lib/gooddata/models/process.rb [hunk 33]
```diff
@@ -191,26 +174,43 @@ def deploy_from_appstore(path, options = { :client => GoodData.client, :project
           }
         }
 
+        save(data, options)
+      end
+
+      def deploy_component(data, options = { :client => GoodData.client, :project => GoodData.project })
+        client, project = GoodData.get_client_and_project(options)
+        data = { process: data } unless data[:process]
+        data[:process] = GoodData::Helpers.symbolize_keys(data[:process]).select { |k| [:type, :name, :component].include? k }
```
Comment (yenkins) [inline lib/gooddata/models/process.rb]: Has the variable name 'k' (UncommunicativeVariableName)

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/85/ : SUCCESS in 10s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/901/ : SUCCESS in 2m 01s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1169/ : SUCCESS in 2m 16s

File: lib/gooddata/models/process.rb [hunk 34]
```diff
@@ -90,16 +90,24 @@ def upload_package(path, files_to_exclude, opts = { :client => GoodData.connecti
       # @option options [String] :name Readable name of the process
       # @option options [String] :process_id ID of a process to be redeployed (do not set if you want to create a new process)
       # @option options [Boolean] :verbose (false) Switch on verbose mode for detailed logging
-      def deploy(path, options = { :client => GoodData.client, :project => GoodData.project })
-        return deploy_brick(path, options) if path.to_s.start_with?(APP_STORE_URL)
-
-        return deploy_from_appstore(path.to_s, options) if (path.to_s =~ %r{\${.*}:(.*)\/(.*):\/}) == 0 # rubocop:disable Style/NumericPredicate
+      def deploy(path, options = { client: GoodData.client, project: GoodData.project })
```
Comment (yenkins) [inline lib/gooddata/models/process.rb]: Takes parameters ['options', 'path'] to 5 methods (DataClump)

File: lib/gooddata/models/process.rb [hunk 35]
```diff
@@ -90,16 +90,24 @@ def upload_package(path, files_to_exclude, opts = { :client => GoodData.connecti
...
+        case path
+        when (path.is_a?(Hash) && path[:component])
+          deploy_component path, options
+        when path.to_s.start_with?(APP_STORE_URL)
```
Comment (yenkins) [inline lib/gooddata/models/process.rb]: Calls 'path.to_s' 3 times (DuplicateMethodCall)

File: lib/gooddata/models/process.rb [hunk 36]
```diff
@@ -191,26 +181,43 @@ def deploy_from_appstore(path, options = { :client => GoodData.client, :project
           }
         }
 
+        save(data, options)
+      end
+
+      def deploy_component(data, options = { client: GoodData.client, project: GoodData.project })
+        client, project = GoodData.get_client_and_project(options)
+        data = { process: data } unless data[:process]
```
Comment (yenkins) [inline lib/gooddata/models/process.rb]: Calls 'data[:process]' 4 times (DuplicateMethodCall)

Comment (yenkins) [inline lib/gooddata/models/process.rb]: Refers to 'data' more than self (maybe move it to another class?) (FeatureEnvy)

File: lib/gooddata/models/process.rb [hunk 37]
```diff
@@ -90,16 +90,24 @@ def upload_package(path, files_to_exclude, opts = { :client => GoodData.connecti
       # @option options [String] :name Readable name of the process
       # @option options [String] :process_id ID of a process to be redeployed (do not set if you want to create a new process)
       # @option options [Boolean] :verbose (false) Switch on verbose mode for detailed logging
-      def deploy(path, options = { :client => GoodData.client, :project => GoodData.project })
-        return deploy_brick(path, options) if path.to_s.start_with?(APP_STORE_URL)
-
-        return deploy_from_appstore(path.to_s, options) if (path.to_s =~ %r{\${.*}:(.*)\/(.*):\/}) == 0 # rubocop:disable Style/NumericPredicate
+      def deploy(path, options = { client: GoodData.client, project: GoodData.project })
+        case path
+        when (path.is_a?(Hash) && path[:component])
```
Comment (yenkins) [inline lib/gooddata/models/process.rb]: Refers to 'path' more than self (maybe move it to another class?) (FeatureEnvy)

File: lib/gooddata/models/process.rb [hunk 38]
```diff
@@ -191,26 +181,43 @@ def deploy_from_appstore(path, options = { :client => GoodData.client, :project
           }
         }
 
+        save(data, options)
+      end
+
+      def deploy_component(data, options = { client: GoodData.client, project: GoodData.project })
```
Comment (yenkins) [inline lib/gooddata/models/process.rb]: Has approx 8 statements (TooManyStatements)

File: lib/gooddata/models/process.rb [hunk 39]
```diff
@@ -90,16 +90,24 @@ def upload_package(path, files_to_exclude, opts = { :client => GoodData.connecti
...
+        when path.to_s.start_with?(APP_STORE_URL)
+          deploy_brick path, options
+        when path.to_s !~ %r{\${.*}:(.*)\/(.*):\/}
+          deploy_from_appstore path.to_s, options
+        else
+          deploy_simple_process path, options
+        end
+      end
 
+      def deploy_simple_process(path, options = { client: GoodData.client, project: GoodData.project })
```
Comment (yenkins) [inline lib/gooddata/models/process.rb]: Has approx 13 statements (TooManyStatements)

File: lib/gooddata/models/process.rb [hunk 40]
```diff
@@ -191,26 +181,43 @@ def deploy_from_appstore(path, options = { :client => GoodData.client, :project
           }
         }
 
+        save(data, options)
+      end
+
+      def deploy_component(data, options = { client: GoodData.client, project: GoodData.project })
+        client, project = GoodData.get_client_and_project(options)
+        data = { process: data } unless data[:process]
+        data[:process] = GoodData::Helpers.symbolize_keys(data[:process]).select { |k| [:type, :name, :component].include? k }
```
Comment (yenkins) [inline lib/gooddata/models/process.rb]: Has the variable name 'k' (UncommunicativeVariableName)

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/86/ : SUCCESS in 10s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/903/ : SUCCESS in 52s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1171/ : SUCCESS in 58s

**Pull request closed without merging**